sdk/samples/iot/paho_iot_hub_methods_sample.c

```diff
@@ -363,30 +212,50 @@ static void receive_messages()
     }
     LOG_SUCCESS("Message #%d: Client received message from the service.", message_count + 1);
 
-    parse_message(topic, topic_len, message, &method_request);
+    // Parse method message and invoke method.
+    az_iot_hub_client_method_request method_request;
+    parse_method_message(topic, topic_len, message, &method_request);
     LOG_SUCCESS("Client parsed message.");
 
     invoke_method(&method_request);
     LOG(" "); // formatting
+
+    MQTTClient_freeMessage(&message);
+    MQTTClient_free(topic);
+  }
+}
+
+void disconnect_mqtt_client_from_iot_hub()
+{
+  int rc;
+
+  if ((rc = MQTTClient_disconnect(mqtt_client, TIMEOUT_MQTT_DISCONNECT_MS)) != MQTTCLIENT_SUCCESS)
+  {
+    LOG_ERROR("Failed to disconnect MQTT client: MQTTClient return code %d.", rc);
+    exit(rc);
   }
 
-  MQTTClient_freeMessage(&message);
-  MQTTClient_free(topic);
-  return;
+  MQTTClient_destroy(&mqtt_client);
 }
 
-static void parse_message(
+void parse_method_message(
```

Comment:
I'm curious for all of these samples why these are now public functions with preconditions.

Reply:
Being a sample, I didn't see the point of using static.  But let's chat.  null checking is still important per comment above.